### DISCUSSION TITLE
When marked.parse return a Promise<string>?

### DISCUSSION THREAD

**Discussion by jcubic** [markedjs/marked#3219] [Q&A] [answered]
The signature of `marked.parse` look like this `(string) => string | Promise<string>` but all documentation suggests it return a string. When It returns a Promise? Is it safe to use:
```typescript
const html = marked.parse('...') as string;
```

**Comment by UziTech** [comment, marked as answer]
An extension can set marked in async mode so it returns a promise. [docs](https://marked.js.org/using_pro#async)

If you don't use extensions, or know none of them are async, it is safe to cast to string.

**Comment by sofiabakke** [comment]
You can send {async: false} as an option to the parse function, and the type will be fixed.
`const html = marked.parse('...', { async: false })`

**Reply by jcubic** [reply]
Good to know, will use it next time I'll work with marked.
